doc/data/messages/c/contextmanager-generator-missing-cleanup/bad.py

```diff
@@ -0,0 +1,14 @@
+import contextlib
+
+
+@contextlib.contextmanager
+def cm():
+    contextvar = "acquired context"
+    print("cm enter")
+    yield contextvar
+    print("cm exit")
+
+
+def genfunc_with_cm():  # [contextmanager-generator-missing-cleanup]
```

Comment:
Why are we raising the message on the function and no on the context manager itself?

Reply:
Yeah, on the issue I suggested that we will want to sniff out only the use cases where it looks like a resource is being managed, AND where our prescription (add cleanup code) makes sense.

You can call a context manager either with a with statement or a decorator. If you call it with a decorator, the cleanup code in the manager won't even make a difference, so we shouldn't warn. And if you call it with a with statement, and you immediately discard the yielded value, it suggests it's not something that needs to be cleaned up.

All of that is easier to tell from the invocation of the manager than from the manager itself, even if it doesn't seem like the purest place to raise the message. Thoughts?

Reply:
Probably worth a short explanation in details.rst and a link out to the python discussion.

Reply:
sorry for slow follow up on this. Should i include a `details.rst` for the message with some more background and reason for the message in `doc/data/`.

Think I would reference the 2 articles you link in the issue and summarizations of those problems, while also contrasting what @DanielNoord points out about generator exhaustion cases

Reply:
Yes please!

Reply:
@jacobtylerwalls added the `details.rst` Let me know if you think I should add anything else there or general grammar and reading